<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [Mapnik-users] Error in python importing mapnik
   </TITLE>
   <LINK REL="Index" HREF="http://lists.berlios.de/pipermail/mapnik-users/2008-November/index.html" >
   <LINK REL="made" HREF="mailto:mapnik-users%40lists.berlios.de?Subject=Re%3A%20%5BMapnik-users%5D%20Error%20in%20python%20importing%20mapnik&In-Reply-To=%3C49243B2C.5070506%40makina-corpus.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="001374.html">
   <LINK REL="Next"  HREF="001382.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mapnik-users] Error in python importing mapnik</H1>
    <B>Gilles Bassi&#232;re</B> 
    <A HREF="mailto:mapnik-users%40lists.berlios.de?Subject=Re%3A%20%5BMapnik-users%5D%20Error%20in%20python%20importing%20mapnik&In-Reply-To=%3C49243B2C.5070506%40makina-corpus.com%3E"
       TITLE="[Mapnik-users] Error in python importing mapnik">gilles.bassiere at makina-corpus.com
       </A><BR>
    <I>Wed Nov 19 17:13:32 CET 2008</I>
    <P><UL>
        <LI>Previous message: <A HREF="001374.html">[Mapnik-users] Error in python importing mapnik
</A></li>
        <LI>Next message: <A HREF="001382.html">[Mapnik-users] Error in python importing mapnik
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1375">[ date ]</a>
              <a href="thread.html#1375">[ thread ]</a>
              <a href="subject.html#1375">[ subject ]</a>
              <a href="author.html#1375">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Hi Vu Hung,

I often had troubles with boost names, it's somewhat confusing I agree.
Regarding the gcc version mention, I believe the BOOST_TOOLKIT has been 
added to scons option to solve this problem.

Regards
Gilles

Nguyen Vu Hung wrote:
&gt;<i> Never mind, I fixed it myself.
</I>&gt;<i>
</I>&gt;<i> At some point, I've rebuilt boost but I didn't rebuild mapnik.
</I>&gt;<i>
</I>&gt;<i> I clean mapnik by running
</I>&gt;<i>
</I>&gt;<i> /usr/bin/python scons/scons.py  -c
</I>&gt;<i>
</I>&gt;<i> and then
</I>&gt;<i>
</I>&gt;<i> /usr/bin/python scons/scons.py
</I>&gt;<i> /usr/bin/python scons/scons.py  install
</I>&gt;<i>
</I>&gt;<i> and it worked.
</I>&gt;<i>
</I>&gt;<i> Btw, under CentOs 5, the libraries generated by boost have *gcc41* in
</I>&gt;<i> their names,
</I>&gt;<i> so I had to do:
</I>&gt;<i>
</I>&gt;<i> cd /home/vuhung/opt/boost/lib
</I>&gt;<i> ln -s libboost_filesystem-gcc41-mt.solibboost_filesystem-mt.so
</I>&gt;<i> ln -s libboost_regex-gcc41-mt.solibboost_regex-mt.so
</I>&gt;<i> ln -s libboost_iostreams-gcc41-mt.solibboost_iostreams-mt.so
</I>&gt;<i> ln -s libboost_program_options-gcc41-mt.solibboost_program_options-mt.so
</I>&gt;<i> ln -s libboost_thread-gcc41-mt.solibboost_thread-mt.so
</I>&gt;<i> ln -s libboost_python-gcc41-mt.solibboost_python-mt.so
</I>&gt;<i>
</I>&gt;<i> If you think this worths to be added to:
</I>&gt;<i> <A HREF="http://trac.mapnik.org/wiki/InstallationTroubleshooting">http://trac.mapnik.org/wiki/InstallationTroubleshooting</A>
</I>&gt;<i> I will do.
</I>&gt;<i>
</I>&gt;<i> On Wed, Nov 19, 2008 at 10:39 PM, Nguyen Vu Hung &lt;<A HREF="https://lists.berlios.de/mailman/listinfo/mapnik-users">vuhung16plus at gmail.com</A>&gt; wrote:
</I>&gt;<i>   
</I>&gt;&gt;<i> Hi all,
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> I've tried to install mapnik and import it from python. Here is what I did:
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> 1. CentOs 5.0, gcc 4.1.2, libstdc++-4.1.2, python  2.4.3( default).
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> 2. boost_1_37_0]$ cat vh-config.sh
</I>&gt;&gt;<i> make clean
</I>&gt;&gt;<i> sh ./configure --prefix=$HOME/opt/boost --with-icu --with-python=/usr/bin/python
</I>&gt;&gt;<i> make
</I>&gt;&gt;<i> make install
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> 3. mapnik-0.5.1]$ cat vh-scons.sh
</I>&gt;&gt;<i> /usr/bin/python scons/scons.py
</I>&gt;&gt;<i> BOOST_INCLUDES=$HOME/opt/boost/include/boost-1_37
</I>&gt;&gt;<i> BOOST_LIBS=$HOME/opt/boost/lib
</I>&gt;&gt;<i> sudo /usr/bin/python scons/scons.py
</I>&gt;&gt;<i> BOOST_INCLUDES=$HOME/opt/boost/include/boost-1_37
</I>&gt;&gt;<i> BOOST_LIBS=$HOME/opt/boost/lib install
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> Then I tried to import mapnik, but ended up with an error.
</I>&gt;&gt;<i> 4. $ /usr/bin/python
</I>&gt;&gt;<i> Python 2.4.3 (#1, May 24 2008, 13:47:28)
</I>&gt;&gt;<i> [GCC 4.1.2 20070626 (Red Hat 4.1.2-14)] on linux2
</I>&gt;&gt;<i> Type &quot;help&quot;, &quot;copyright&quot;, &quot;credits&quot; or &quot;license&quot; for more information.
</I>&gt;&gt;<i>     
</I>&gt;&gt;&gt;&gt;&gt;<i> from mapnik import *
</I>&gt;&gt;&gt;&gt;&gt;<i>           
</I>&gt;&gt;<i> Traceback (most recent call last):
</I>&gt;&gt;<i>  File &quot;&lt;stdin&gt;&quot;, line 1, in ?
</I>&gt;&gt;<i>  File &quot;/usr/lib/python2.4/site-packages/mapnik/__init__.py&quot;, line 31, in ?
</I>&gt;&gt;<i>    from _mapnik import *
</I>&gt;&gt;<i> ImportError: /home/vuhung/opt/boost/lib/libboost_python-gcc41-mt-1_37.so.1.37.0:
</I>&gt;&gt;<i> undefined symbol: _Py_RefTotal
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> I am new to both boost, python and mapnik.
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> Any ideas?
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>
</I>&gt;&gt;<i> --
</I>&gt;&gt;<i> Best Regards,
</I>&gt;&gt;<i> Nguyen Hung Vu ( Nguy&#7877;n V&#361; H&#432;ng )
</I>&gt;&gt;<i> vuhung16plus{remove}@gmail.dot.com , YIM: vuhung16 , Skype: vuhung16dg
</I>&gt;&gt;<i> A brief profile: <A HREF="http://www.hn.is.uec.ac.jp/~vuhung/Nguyen.Vu.Hung.html">http://www.hn.is.uec.ac.jp/~vuhung/Nguyen.Vu.Hung.html</A>
</I>&gt;&gt;<i>
</I>&gt;&gt;<i>     
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i>   
</I>
-- 
Gilles Bassiere
MAKINA CORPUS
30 rue des Jeuneurs
FR-75002 PARIS
+33 (0) 1 44 82 00 80
<A HREF="http://www.makina-corpus.com">http://www.makina-corpus.com</A>

-------------- next part --------------
A non-text attachment was scrubbed...
Name: gilles_bassiere.vcf
Type: text/x-vcard
Size: 380 bytes
Desc: not available
URL: &lt;<A HREF="https://lists.berlios.de/pipermail/mapnik-users/attachments/20081119/0d1fb1e9/attachment.vcf">https://lists.berlios.de/pipermail/mapnik-users/attachments/20081119/0d1fb1e9/attachment.vcf</A>&gt;
</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="001374.html">[Mapnik-users] Error in python importing mapnik
</A></li>
	<LI>Next message: <A HREF="001382.html">[Mapnik-users] Error in python importing mapnik
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1375">[ date ]</a>
              <a href="thread.html#1375">[ thread ]</a>
              <a href="subject.html#1375">[ subject ]</a>
              <a href="author.html#1375">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.berlios.de/mailman/listinfo/mapnik-users">More information about the Mapnik-users
mailing list</a><br>
</body></html>
